test: update k8s versions to 1.9.10 1.10.7 1.11.2 and 1.12.0-beta.0

aanm opened the pull request:
Signed-off-by: André Martins <andre@cilium.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5328)
<!-- Reviewable:end -->

aanm committed:
test: update k8s versions to 1.9.10 1.10.7 1.11.2 and 1.12.0-beta.0

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
test: re-add k8s 1.12 to CI

Signed-off-by: André Martins <andre@cilium.io>

ianvernon commented on ginkgo-kubernetes-all.Jenkinsfile:
```diff
@@ -145,6 +149,7 @@ pipeline {
             sh "cd ${TESTDIR}; K8S_VERSION=1.8 vagrant destroy -f || true"
```
Does this mean that we are only destroying the VMs after the entire pipeline has run? I.e., the K8S 1.8 and 1.9 VMs are still running at the start of the stage for K8s 1.10 and K8s 1.12?

aanm replied:
cc @eloycoto

eloycoto replied:
https://github.com/cilium/cilium/blob/0c9728feab4f3354634e8db07512ac4b72a66137/test/test_suite_test.go#L245-L262

ianvernon replied:
Great, thanks @eloycoto !

eloycoto commented on test/provision/k8s_install.sh:
```diff
@@ -162,8 +182,15 @@ sudo iptables --policy FORWARD ACCEPT
 
 #check hostname to know if is kubernetes or runtime test
 if [[ "${HOST}" == "k8s1" ]]; then
+    case $K8S_VERSION in
```
Due that we already have versions for this, and to avoid this if, why we don't have a $KUBEADM_CONFIG, and by default is `KUBEADM_CONFIG=$KUBEADM_CONFIG_ALPHA1` and in the 1.12 option you change to $KUBEADM_CONFIG as `$KUBEADM_CONFIG_ALPHA2`

Like the DNS deployment.

aanm replied:
Done

eloycoto commented on test/provision/k8s_install.sh:
```diff
@@ -77,6 +77,26 @@ networking:
+networking:
+  dnsDomain: cluster.local
+  podSubnet: "{{ .KUBEADM_POD_NETWORK }}/{{ .KUBEADM_POD_CIDR}}"
+  serviceSubnet: 10.96.0.0/12
```
Quick one, in case that we have IPv6, this still works? or need to have a different value?

aanm replied:
@eloycoto We need to have an IPv6 specified here.

eloycoto replied:
Ok, could you fix this now? Or add a note that need to be change?

aanm replied:
@eloycoto Fix what? Are we testing k8s with IPv6 in the CI?

eloycoto replied:
No, but there is a option for running test on IPv6, and If it's break I would like to have a note in there.

aanm replied:
Done

aanm committed:
examples/kubernetes: add node.kubernetes.io/not-ready toleration

Kubernetes 1.12 added node.kubernetes.io/not-ready for nodes that don't
have its runtime network ready. Since Cilium needs to be deployed on
nodes so it can setup the CNI configuration the not-ready toleration
needs to be added to the DaemonSet.

Signed-off-by: André Martins <andre@cilium.io>
